### PR TITLE
feat: don't open action-url if notification onClick handler returns false

### PR DESCRIPTION
**Pull request by pianomansam**
Fixes https://github.com/magicbell-io/magicbell-js/issues/61

### CODE REVIEW

File: packages/react/src/components/ClickableNotification/ClickableNotification.tsx [hunk 1]
```diff
@@ -40,12 +40,12 @@ export default function ClickableNotification({ notification: rawNotification, o
...
 
     if (isActionableElement) return;
-    if (onClick) onClick(notification);
+    const onClickResult = onClick?.(notification) || true;
     if (notification.actionUrl) {
       // We wait for the markAsRead before navigating to the new url, to prevent race conditions
       // between mark and read, and fetching new data on page reload. But let's not wait forever.
       const timeout = new Promise((resolve) => setTimeout(resolve, 1_000));
-      Promise.race([markAsReadPromise, timeout]).then(() => openActionUrl(notification));
+      Promise.race([markAsReadPromise, timeout]).then(() => onClickResult && openActionUrl(notification));
```
Comment (smeijer) [inline packages/react/src/components/ClickableNotification/ClickableNotification.tsx]:
```suggestion
      Promise.race([markAsReadPromise, timeout]).then(() => onClickResult !== false && openActionUrl(notification));
```

**Review by smeijer — APPROVED** [review]
Sweet! Thanks.

Can you also run `yarn changeset`, flag the react package as `patch`, and commit those files? Then our CI will pick up the change and publish a new version. 🙂

If not, no problem. I can do it later.

**Comment by pianomansam** [comment]
@smeijer I think I'll let you handle the `yarn changeset` workflow as I'm not familiar with how you have it all set up.

**Comment by smeijer** [comment]
This has been released as [@magicbell/magicbell-react@10.5.0](https://github.com/magicbell-io/magicbell-js/releases/tag/@magicbell/magicbell-react@10.5.0) and [@magicbell/embeddable@3.2.9](https://github.com/magicbell-io/magicbell-js/releases/tag/@magicbell/embeddable@3.2.9).

Thanks again for the contribution 🙂

**Pull request merged**